sdk/attestation/azure-security-attestation/azure/security/attestation/_models.py

```diff
@@ -880,99 +859,92 @@ def _create_unsecured_jwt(body):
             pass
-        json_body = ''
+        json_body = ""
         if body is not None:
-            json_body = JSONEncoder().encode(body)
-        return_value += '.'
-        return_value += Base64Url.encode(json_body.encode('utf-8'))
+            json_body = json.dumps(body)
+        return_value += "."
+        return_value += base64url_encode(json_body.encode("utf-8"))
 
         # Now we want to sign the return_value.
-        if isinstance(signer._signing_key, RSAPrivateKey):
-            signature = signer._signing_key.sign(
-                return_value.encode('utf-8'),
+        if isinstance(key, RSAPrivateKey):
+            signature = key.sign(
+                return_value.encode("utf-8"),
                 algorithm=SHA256(),
-                padding=padding.PKCS1v15())
+                padding=padding.PKCS1v15(),
+            )
         else:
-            signature = signer._signing_key.sign(
-                return_value.encode('utf-8'),
-                algorithm=SHA256())
+            signature = key.sign(return_value.encode("utf-8"), algorithm=SHA256())
         # And finally append the base64url encoded signature.
-        return_value += '.'
-        return_value += Base64Url.encode(signature)
+        return_value += "."
+        return_value += base64url_encode(signature)
         return return_value
 
-class AttestationTokenValidationException(Exception):
-    """ Thrown when an attestation token validation fails.
 
-    :param str message: Message for caller describing the reason for the failure.
+class AttestationPolicyToken(AttestationToken):
     """
-    def __init__(self, message):
-        self.message = message
-        super(AttestationTokenValidationException, self).__init__(self.message)
-        
+    An `AttestationPolicyToken` is an AttestationToken object specialized for
+    use in attestation policy :func:`AttestationAdministrationClient.set_policy`
+    operations. It expresses the JSON Web Signature object sent to the
+    attestation service to set the token.
+
+    This token can thus be used to validate the hash returned by the `set_policy` API.
 
-class AttestationResponse(Generic[T]):
-    """ Represents a response from the attestation service.
+    :param str policy: Attestation Policy to be used in the body of the token.
 
-    :param token: Attestation Token returned from the service.
-    :type token: azure.security.attestation.AttestationToken
-    :param value: Value of the body of the attestation token.
-    :type value: T
     """
-    def __init__(self, token, value):
-        # type (AttestationToken, T) -> None
-        self.token = token #type: AttestationToken
-        self.value = value #type: T
 
-class TpmAttestationRequest(object):
-    """ Represents a request for TPM attestation.
+    def __init__(self, policy, **kwargs):
+        # type: (str, Dict[str, Any]) -> None
+        AttestationToken.__init__(self, body=StoredAttestationPolicy(policy), **kwargs)
```

Comment:
```suggestion
        super(  # pylint: disable=super-with-arguments
            AttestationPolicyToken, self
        ).__init__(self, body=StoredAttestationPolicy(policy), **kwargs)
```

Reply:
I thought that super was considered bad practice and should be avoided. Is that not the case?

Reply:
In my experience (working on Key Vault and Identity), we use super consistently. I notice that our Python guidelines don't actually seem to have any guidance on this (which may be something we should take a look at, since there can be issues from inconsistent use) -- but since using super is a convention in at least some libraries and, as that link points out, "For correct behavior, if any class uses super, all its subclasses must also", it makes sense to continue using it.

Reply:
Makes sense, I switched it to use super (minor nit to your suggestion: The `self` parameter to __init__ doesn't get passed into the `__init__` routine. Not 100% sure why, probably related to the parameters to super.